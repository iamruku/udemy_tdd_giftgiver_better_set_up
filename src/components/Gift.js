import React, { Component } from 'react';

export default class Gift extends Component {
	state = { person: '', gift: '' };

	onRemoveGift = () => {
		this.props.removeGift(this.props.gift.id);
	};

	render() {
		return (
			<div>
				<form className="cls-form">
					<label>Name</label>
					<input
						type="text"
						className="inp-person"
						onChange={e => this.setState({ person: e.target.value })}></input>
					<label>Gift</label>
					<input
						type="text"
						className="inp-gift"
						onChange={e => this.setState({ gift: e.target.value })}></input>
				</form>
				<button className="btn-delete" onClick={this.onRemoveGift}>
					Remove Gift
				</button>
			</div>
		);
	}
}
