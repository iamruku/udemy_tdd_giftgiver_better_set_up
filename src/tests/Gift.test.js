import React from 'react';
import { shallow } from 'enzyme';
import Gift from '../components/Gift';
import './setupTests';

describe('Gift', () => {
	const _removeGift = jest.fn();
	const id = 1;
	const props = { gift: { id }, removeGift: _removeGift };
	const wrapper = shallow(<Gift {...props}></Gift>);

	describe('initial dom content', () => {
		it('renders wrapper', () => {
			expect(wrapper).toMatchSnapshot();
		});

		it('should have an input field for name', () => {
			expect(
				wrapper
					.find('.cls-form')
					.find('.inp-person')
					.exists()
			).toBe(true);
		});

		it('should have an input field for gift', () => {
			expect(
				wrapper
					.find('.cls-form')
					.find('.inp-gift')
					.exists()
			).toBe(true);
		});

		it('should have a remove gift button', () => {
			expect(wrapper.find('.btn-delete').exists()).toBe(true);
		});
	});

	describe('initial state content', () => {
		it('should initialize person to empty', () => {
			expect(wrapper.state().person).toEqual('');
		});
		it('should initialize gift to empty', () => {
			expect(wrapper.state().gift).toEqual('');
		});
	});

	describe('value change in person input', () => {
		const personString = 'test_person';

		beforeEach(() => {
			wrapper
				.find('.inp-person')
				.simulate('change', { target: { value: personString } });
		});

		it('updates the person in `state`', () => {
			expect(wrapper.state().person).toEqual(personString);
		});
	});

	describe('value change in gift input', () => {
		const giftString = 'test_gift';

		beforeEach(() => {
			wrapper
				.find('.inp-gift')
				.simulate('change', { target: { value: giftString } });
		});

		it('updates the gift in `state`', () => {
			expect(wrapper.state().gift).toEqual(giftString);
		});
	});

	describe('click on delete gift', () => {
		beforeEach(() => {
			wrapper.find('.btn-delete').simulate('click');
		});

		it('should call the removeGift callback with id', () => {
			expect(_removeGift).toHaveBeenCalledWith(id);
		});
	});
});
