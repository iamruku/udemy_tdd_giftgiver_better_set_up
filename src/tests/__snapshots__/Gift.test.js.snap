// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift initial dom content renders wrapper 1`] = `ShallowWrapper {}`;
